### PR TITLE
Add more worked examples, including linear regression and an MLP MNIST

### PR DESCRIPTION
**Pull request by staticfloat**
I think it is worthwhile to have more examples showing users what usage of Zygote can look like, as it is just so incredibly clean and beautiful.

### CODE REVIEW

**Comment by oxinabox** [comment]
Do we have something that autobakes this into the docs?

**Comment by staticfloat** [comment]
> Do we have something that autobakes this into the docs?

I'm not sure what this means.  But we should have some kind of CI for this, because I just discovered that the `profiler.jl` example has bitrotted, and I'm not sure how to fix it.  :P

**Comment by oxinabox** [comment]
It would be nice if the code examples,
from this file

Showed up inside the documention website (with syntax highlighting etc)

File: examples/linear_regression.jl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+# of our model a tad bit according to the gradients
+function sgd_update!(model::LinearRegression, grads, η = 0.001)
+    model.weights .-= η .* grads.weights
+    model.bias -= η * grads.bias
+end
+
+# Now let's do that for each example in our training set:
+@info("Running train loop for $(size(X,2)) iterations")
+for idx in 1:size(X, 2)
+    grads = Zygote.gradient(m -> loss(m, X[:, idx], Y[idx]), model)[1][]
```
Comment (JeffBezanson) [inline examples/linear_regression.jl]: I think the `[]` is not needed here?

Comment (staticfloat) [reply examples/linear_regression.jl]: The `[]` is because `mutable struct` objects get wrapped in a `Ref` when passed back from Zygote.

Comment (JeffBezanson) [reply examples/linear_regression.jl]: Oops, that's only if you make it immutable.

**Comment by MikeInnes** [comment]
Really nice examples, thanks!

**Pull request merged**